src/Sylius/Bundle/CoreBundle/Applicator/CatalogPromotionApplicator.php

```diff
@@ -93,6 +97,9 @@ private function applyDiscountFromActionOnChannelPricing(
 
         $channelPricing->setPrice($price);
         $channelPricing->addAppliedPromotion($this->appliedPromotionInformationFormatter->format($catalogPromotion));
+        if ($catalogPromotion->isExclusive()) {
+            $channelPricing->setExclusiveCatalogPromotionApplied(true);
```

Comment:
Maybe it would be better to include this information in the `appliedPromotions` array - additional field looks weird, as we already have all the information about applied discounts gathered together